app/post_test.go

```diff
@@ -668,6 +668,35 @@ func TestCreatePost(t *testing.T) {
 		require.Nil(t, err)
 		assert.Equal(t, "![image]("+proxiedImageURL+")", rpost.Message)
 	})
+
+	t.Run("Sets POST_PROPS_MENTION_HIGHLIGHT_DISABLED and sends an ephemeral post when user does not have required permission", func(t *testing.T) {
```

Comment:
Yea i wanted to test it - although im not exactly sure how to go about testing that since nothing gets committed to the db.

I'm not fluent at the right way to mock for testing in Go but from what I can see the approach seems to be that the methods should be defined on an interface in this case an AppIface. Allowing for a test struct to be created where specific methods can be mocked (in this case I would mock `SendEphemeralPost` to have a way to check that it was called)
But our `helper_test` setup uses the `struct` `App` and not an Interface and all of the app methods are defined on the `struct`. So im not sure how I would go about it using this approach without a major refactoring of the App.

Let me know if im overcomplicating this and if theres an easier way to do this... 
I guess I might be able to listen for the websocket event to see if a post event was sent

Reply:
I was thinking about it in a more traditional TDD way - this behaviour cannot be unit tested here so I added tests to the post API asserting that the ephemeral message is sent to the client